builder/rootfs/bin/boot

```diff
@@ -74,6 +88,27 @@ done
 docker build -t deis/slugbuilder /usr/local/src/slugbuilder/
 docker build -t deis/slugrunner /usr/local/src/slugrunner/
 
+function gen_host_keys {
+	# generate the keys, then set them up in etcd if not already set
+	/usr/bin/ssh-keygen -A
```

Comment:
The logic follows like so:

 - generate new keys on every boot
 - set keys in etcd *if not already set*
 - read keys from etcd, write to the local filesystem

In essence, the first time the keys are generated they are pushed to etcd. After a reboot, the keys will attempt to write but hit the "Key not found" error, making it a no-op. The keys that were previously set in etcd are now read and written to the local filesystem.